sdk/servicebus/service-bus/src/receivers/receiver.ts

```diff
@@ -136,53 +137,63 @@ export interface Receiver<ReceivedMessageT> {
  */
 export class ReceiverImpl<ReceivedMessageT extends ReceivedMessage | ReceivedMessageWithLock>
   implements Receiver<ReceivedMessageT> {
-  /**
-   * @property Describes the amqp connection context for the QueueClient.
-   */
-  private _context: ClientEntityContext;
   private _retryOptions: RetryOptions;
   /**
    * @property {boolean} [_isClosed] Denotes if close() was called on this receiver
    */
   private _isClosed: boolean = false;
 
-  public entityPath: string;
+  /**
+   * Instance of the BatchingReceiver class to use to receive messages in pull mode.
+   */
+  private _batchingReceiver?: BatchingReceiver;
+
+  /**
+   * Instance of the StreamingReceiver class to use to receive messages in push mode.
+   */
+  private _streamingReceiver?: StreamingReceiver;
 
   /**
    * @throws Error if the underlying connection is closed.
    */
   constructor(
-    context: ClientEntityContext,
+    private _context: ConnectionContext,
+    public entityPath: string,
     public receiveMode: "peekLock" | "receiveAndDelete",
     retryOptions: RetryOptions = {}
   ) {
-    throwErrorIfConnectionClosed(context.namespace);
-    this.entityPath = context.entityPath;
-    this._context = context;
+    throwErrorIfConnectionClosed(_context);
     this._retryOptions = retryOptions;
+    if (!this._context.managementClients[entityPath]) {
+      this._context.managementClients[entityPath] = new ManagementClient(
```

Comment:
Let's create a function in _context to create the management client (or we can just create it on-demand). As it is currently we have two spots that create it (sender and receiver).

Reply:
Done with https://github.com/Azure/azure-sdk-for-js/pull/9710/commits/ed41c32649fd20ca340d73eb8cd032660dc06a15